File: Python Fundamentals/Mid Exam problems/mu_online.py
rooms = input().split('|')

hp = 100
bitcoins = 0
passed_rooms = 0

for current_room in rooms:
    passed_rooms += 1
    data = current_room.split()
    if data[0] == 'potion':
        if hp + int(data[1]) > 100:
            healed = 100 - hp
            hp = 100
        else:
            hp += int(data[1])
            healed = int(data[1])
        print(f"You healed for {healed} hp.")
        print(f"Current health: {hp} hp.")
    elif data[0] == 'chest':
        print(f"You found {int(data[1])} bitcoins.")
        bitcoins += int(data[1])
    else:
        hp -= int(data[1])
        if hp > 0:
            print(f"You slayed {data[0]}.")
        else:
            print(f"You died! Killed by {data[0]}.")
            print(f"Best room: {passed_rooms}")
            exit()

print("You've made it!")
print(f"Bitcoins: {bitcoins}")
print(f"Health: {hp}")

File: JavaScript Fundamentals/Mid Exams/Mid Exam 5/muOnline.js
function solve(string) {
    let arr = string.split('|');
    let health = 100;
    let bitcoins = 0;
    let rooms = 0;
    let lose = false;

    for (let element of arr) {
        rooms++;
        let tokens = element.split(' ');
        if (tokens[0] == 'potion') {
            let hp = Number(tokens[1]);
            let healed = 0;
            if ((health + hp) <= 100) healed = hp;
            else healed = 100 - health;
            health += healed;
            console.log(`You healed for ${healed} hp.`);
            console.log(`Current health: ${health} hp.`);
        } else if (tokens[0] == 'chest') {
            console.log(`You found ${tokens[1]} bitcoins.`);
            bitcoins += Number(tokens[1]);
        } else {
            health -= Number(tokens[1]);
            if (health > 0) console.log(`You slayed ${tokens[0]}.`);
            else {
                console.log(`You died! Killed by ${tokens[0]}.`);
                console.log(`Best room: ${rooms}`);
                lose = true;
                break;
            }
        }
    }

    if (!lose) {
        console.log("You've made it!");
        console.log(`Bitcoins: ${bitcoins}`);
        console.log(`Health: ${health}`);
    }
}
